aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -200,6 +200,69 @@ void index_fill_kernel(
   });
 }
 
+void index_copy_kernel(
+  TensorIterator& iter,
+  int64_t dim,
+  int64_t self_dim_size,
+  int64_t self_dim_stride) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(ScalarType::Half, ScalarType::Bool, ScalarType::BFloat16,
+    iter.dtype(), "index_copy_cpu", [&] {
+    auto handle_nonzero_idx_stride = [&](char** data, const int64_t* strides, int64_t n) {
+      auto* self_data_bytes = data[0];
+      auto* index_data_bytes = data[1];
+      auto* source_data_bytes = data[2];
+      for (int64_t elem = 0; elem < n; ++elem) {
+        auto* self_data = reinterpret_cast<scalar_t*>(self_data_bytes);
+        auto idx = *reinterpret_cast<int64_t*>(index_data_bytes);
+        auto* source_data = reinterpret_cast<scalar_t*>(source_data_bytes);
+        TORCH_CHECK_INDEX(idx >= 0 && idx < self_dim_size,
+              "index ", idx, " is out of bounds for dimension ",
+              dim, " with size ", self_dim_size);
+
+        self_data[idx * self_dim_stride] = *source_data;
+
+        self_data_bytes += strides[0];
+        index_data_bytes += strides[1];
+        source_data_bytes += strides[2];
+      }
+    };
+    auto handle_zero_idx_stride = [&](char** data, const int64_t* strides, int64_t n) {
+      auto* self_data_bytes = data[0];
+      auto* index_data_bytes = data[1];
+      auto* source_data_bytes = data[2];
+      auto idx = *reinterpret_cast<int64_t*>(index_data_bytes);
+      if (idx < -self_dim_size || idx >= self_dim_size) {
```

Comment:
Doesn't this index still need to be fixed? 

Also, these error messages should start with the name of the operation (index_copy_, like in the above error messages).

Reply:
Absolutely. I have fixed these two things now.